Defend rebrand from Egress to KnowBe4

olliespires opened the pull request:
Required items, please complete
   
   Change(s):
   - Changed display wording from Egress to KnowBe4
   - Changed file names and folders to match

   Reason for Change(s):
   - Egress has been purchased by KnowBe4

   Version Updated:
   - Yes, from 3.0.0 to 3.1.0, was this correct?

   Testing Completed:
   - Yes
   - Tested using the V3 solution builder
   - Tested in our Sentinel Dev environment, see below screenshot of events coming through
<img width="1068" height="545" alt="image" src="https://github.com/user-attachments/assets/06373771-45a9-4de4-ab70-d4b2f34034ff" />

   - Also tested with the UI definition sandbox as shown below, can see KnowBe4 Defend rather than Egress Defend wording:
<img width="489" height="208" alt="image" src="https://github.com/user-attachments/assets/427e8fe4-0908-4607-95fb-efba04f38662" />
 

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes (check, will finish this when they have run)



> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

olliespires commented:
> @olliespires the command you issued was incorrect. Please try again.
> 
> Examples are:
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> and
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree company="KnowBe4"

olliespires commented:
Hi @v-maheshbh, I have copied the KnowBe4 logo into the workbooks logos folder, correct me if I'm wrong but from the error I think that should fix the failing Workbooks validation. 

I also saw the KQL validation failed, should I be removing the DefendAuditData.txt from the parsers folder? Will this affect anything?

v-maheshbh commented:
Hi @olliespires 

BasePath Format:
Modify the BasePath to the following structure:
C:\GitHub\Azure-Sentinel\solutions\Solution Name
Release Notes Version:
Update the version in the ReleaseNotes to 3.0.1
Ensure that the latest version is listed first in the release notes and include an appropriate comment.
and please refer to the detailed solution and the corresponding folder structure below for further clarity
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoMeraki

Thanks!

v-maheshbh commented:
Hi @olliespires 

Kindly review the comments mentioned above.

Thanks!

v-maheshbh commented:
Hi @olliespires 

Kindly update the file extension, as only .yaml is allowed under the Parsers folder.
<img width="255" height="162" alt="image" src="https://github.com/user-attachments/assets/ff95ce33-364b-4c59-b41d-b4c12e74e6f3" />

and renamed the workbook preview images in the below‑mentioned path as well. Also, kindly check the release notes and remove any extra dotted lines.
https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/Images/Preview


Thanks!

olliespires commented:
> Hi @olliespires
> 
> Kindly update the file extension, as only .yaml is allowed under the Parsers folder. <img alt="image" width="255" height="162" src="https://private-user-images.githubusercontent.com/207855009/533386558-ff95ce33-364b-4c59-b41d-b4c12e74e6f3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc4ODg2MzYsIm5iZiI6MTc2Nzg4ODMzNiwicGF0aCI6Ii8yMDc4NTUwMDkvNTMzMzg2NTU4LWZmOTVjZTMzLTM2NGItNGM1OS1iNDFkLWI0YzEyZTc0ZTZmMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwOFQxNjA1MzZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOTg1NTVmM2Y2N2E3NTNlYTY4YWNiMDJmMTIwMTY2MWI0NzUyN2Q2NTRiMTYyZWE1YWYyN2YzNDAzN2NhZDZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.z5wpU3IMYQIJyd6hja8PZ63yt28ih8UihAuD7ucdNfg">
> 
> and renamed the workbook preview images in the below‑mentioned path as well. Also, kindly check the release notes and remove any extra dotted lines. https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/Images/Preview
> 
> Thanks!

Hi @v-maheshbh ,

Thanks, I have removed that text file, was this from the old packaging flow?
Sorry not sure what you mean about the rename, we have intentionally renamed those logos as part of the rebrand, will that not work?
I have checked the release notes and they look ok to me unless I'm missing something

Cheers!

v-maheshbh commented:
Hi @olliespires 
As you have renamed the workbook preview images under the solution’s Images folder, kindly apply the same renaming to the items mentioned below.  This will resolve the workbook metadata validation failure,

<img width="879" height="254" alt="image" src="https://github.com/user-attachments/assets/6710a6d4-33be-4680-bb7c-507f8ea990fc" />

<img width="664" height="241" alt="image" src="https://github.com/user-attachments/assets/8d7da5e5-6dc6-46e0-bba0-411863ae3f09" />

Thanks!

olliespires commented:
> Hi @olliespires
> 
> Kindly update the parser file extension to .yaml in the data file.
> 
> <img alt="image" width="330" height="113" src="https://private-user-images.githubusercontent.com/207855009/533790291-f52c24a3-8a8a-4689-a84a-8fdd68408cad.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njc5NDczMjMsIm5iZiI6MTc2Nzk0NzAyMywicGF0aCI6Ii8yMDc4NTUwMDkvNTMzNzkwMjkxLWY1MmMyNGEzLThhOGEtNDY4OS1hODRhLThmZGQ2ODQwOGNhZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwOVQwODIzNDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OTRiNzJhMzRkMGVhMDEwZjNlYWMyZDM5MTc0YTg5MGEyNTU3ZTlmNDJlNjE0ZTQxMjVlN2UxMTg2ZTRlYmJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sVXNmE86uzvrYpwMPszUOYG6BtODto27qq7GCclkh2s">
> Thanks!

Hi @v-maheshbh ,

Nice, updated that 

Thanks!
